src/chainlit/langchain/callbacks.py

```diff
@@ -23,7 +27,51 @@ def get_llm_settings(invocation_params: Union[Dict, None]):
         model_name = invocation_params.pop("model_name")
         invocation_params["model"] = model_name
 
-    return provider, invocation_params
+    merged = {**invocation_params, **serialized.get("kwargs", {})}
+
+    # make sure there is no api key specification nor _type
+    settings = {
+        k: v for k, v in merged.items() if not k.endswith("_api_key") and k != "_type"
+    }
+
+    # Automatically add the provider if it doesn't exist yet
+    if not has_llm_provider(provider):
+        path = serialized.get("id", [])
+        provider_name = path[-1]
+        provider_path = ".".join(path[:-1])
+
+        module = importlib.import_module(provider_path)
+        llm_class = getattr(module, provider_name)
+        llm = llm_class()
```

Comment:
i think we should instantiate the llm with the current settings, including the API key. If the API key was passed to the llm and is not available in the env variables this won't work

Reply:
The api key isn't available there, so we will keep instantiating the LLM without parameters for now. User can still declare their own (non automated) provider to go around this issue.